### ISSUE TITLE
[bug] command may get stuck when wait for state lock and may confusing args type, and `pnpm tauri info` will failed now

### ISSUE THREAD

**Issue by Hellager** [tauri-apps/tauri#6566]
### Describe the bug

Here's my reproduction repo: [problems-for-tauri-template](https://github.com/Hellager/problems-for-tauri-template)

1. Command stuck when wait for state lock:
After running command `start_cron_job`, try invoke command `stop_cron_job` will stuck at `state.0.lock().await`, other commands works fine.

2. Command confusing its args type
My command `update_filter_blacklist` has this definition
`pub async fn update_filter_blacklist(data: Vec<CleanFilterItem>, state: State<'_, CleanRecentHandlerWrapper>)`
, and its works well when give it an empty vec.
But when i try give it a vec with some data, it shows 
`invalid args 'data' for command 'update_filter_blacklist': invalid type: integer '0', expected string or map`.
Seems like the command be a little confusing about its args?

3.  `pnpm tauri info` failed now
As it says it will panic for this error:
`thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected character 't' while parsing major version number")', src\info.rs:566:67`,
and my os version should be win11 22h2, not win10

### Reproduction

1. Clone repo [problems-for-tauri-template](https://github.com/Hellager/problems-for-tauri-template).
2. Run `pnpm install && pnpm tauri dev`.
3. Click `Start Cron Job` and then click `Stop Cron Job`, you will see it stuck at `state.0.lock().await`.
4. Click `Build Masked List` then you will see the invalid args error.
5. Run `pnpm tauri info` will get the unexcepted character 't' error

### Expected behavior

1. command should not be stuck at `state.0.lock().await` if others not stucked.
2. with correct given type of args, command should not get invalid args error.
3. pnpm tauri info should work fine

### Platform and versions

```text
Environment
  › OS: Windows 10.0.22621 X64
  › Webview2: 111.0.1661.51
  › MSVC: 
      - Visual Studio Community 2022
  › Node.js: 16.15.1
  › npm: 8.15.0
  › pnpm: 7.28.0
  › yarn: 1.22.19
  › rustup: 1.25.2
  › rustc: 1.68.1
  › cargo: 1.68.1
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected character 't' while parsing major version number")', src\info.rs:566:67
```

**Comment by FabianLars** [comment]
1. I think this is a code logic mistake, or a deadlock. You're locking the Mutex twice in start_cron_job so to me it looks like it's forever waiting on the line of the second lock() call, therefore the _third_ lock() call in stop_cron_job can never acquire a lock either.
2. My guess is that it fails on the level enum 🤔 typescript tends to convert enums to numbers which serde (on the rust side) doesn't understand by default afaik
3. That should be fixed in v1.3

**Comment by Hellager** [comment]
Thanks for your help!
1. Sorry for missing the lock action in command, that's the real probelm.
2. It did fails because of enum, i have to set enum in rust to u16 to solve the probem.
